### PR TITLE
Fix case in Typescript code for pattern_useconstantlod parameter to match backend code.

### PR DESCRIPTION
**Pull request by MarcNeely**
There is an inconsistency between the name of the pattern_useconstantlod parameter in TextureMapProps between the Typescript code and the backend code.  This PR makes it match so that modifying the property in Typescript will actually have an effect.

Fixes https://github.com/orgs/iTwin/discussions/23

### CODE REVIEW

**Review by pmconne — APPROVED** [review]

**Comment by pmconne** [comment]
@Mergifyio backport release/4.4.x

**Comment by pmconne** [comment]
@aruniverse @DanRod1999 fyi

**Comment by aruniverse** [comment]
> @aruniverse @DanRod1999 fyi

fyi @anmolshres98

File: core/common/src/MaterialProps.ts [hunk 1]
```diff
@@ -63,14 +63,14 @@ export interface TextureMapProps {
   /** Weight at which to combine diffuse image and color; if undefined, defaults to 1.0 */
   pattern_weight?: number;
   /** If true, override the mapping mode with constant LOD mapping for the normal map, defaults to false. */
-  pattern_useConstantLod?: boolean;
-  /** The number of times the texture is repeated if pattern_useConstantLod is true.  Increasing this will make the texture pattern appear smaller, decreasing it will make it larger. Defaults to 1.*/
+  pattern_useconstantlod?: boolean;
```
Comment (pmconne) [inline core/common/src/MaterialProps.ts:66]: The camel-cased one is `@public` which means you can't delete it without breaking people's code, which we can't do in a minor release.
Mark it `@deprecated in 4.4. It never functioned properly - use [[pattern_useconstantlod]] instead`.

Comment (MarcNeely) [reply core/common/src/MaterialProps.ts:70]: done.

**Review by pmconne — APPROVED** [review]

**Pull request merged**